[randallnhr] iP

randallnhr opened the pull request:
# DukeBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

## Where do I begin?
All you need to do is,
1. download the JAR file from ([here](https://github.com/randallnhr/ip/releases/tag/A-Jar)).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

## Features

- [x] Managing tasks
- [ ] Managing deadlines \(coming soon)
- [ ] Reminders \(coming soon)

Commit:
Add Gradle support

randallnhr committed:
Level-1

randallnhr committed:
Level-1

randallnhr committed:
Level-2

randallnhr committed:
Level-3

randallnhr committed:
Level-4

randallnhr committed:
A-TextUiTesting

randallnhr committed:
Level-5

randallnhr committed:
Level-6

runoutofit commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,20 @@
+    protected String startDate;
+    protected String endDate;
+
+    public Task(String name, String date1, String date2) {
```
would it be better to abstract out startDate and endDate to Event class?

runoutofit commented on src/main/java/Duke.java:
```diff
@@ -12,8 +13,9 @@ public static void main(String[] args) throws IOException {
         BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
 
-        String[] word = br.readLine().split(" ");
+        String[] word = br.readLine().split(" ",2);
```
i like that you store word as a String array for easier manipulation

runoutofit replied:
would renaming word make it less vague and reflect more as a command?

randallnhr committed:
Level-7

randallnhr committed:
Level-8

randallnhr committed:
updated input and expected txt files

randallnhr committed:
Merge branch 'branch-Level-8'

randallnhr committed:
Merge Level-8 to master

randallnhr committed:
A-MoreOOP

antonlee59 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    @Override
+    public String toString() {
+        return this.isDone
+                ? "[D][X] " + this.name + " (by: " + getDate(this.deadline) + ")"
+                : "[D][ ] " + this.name + " (by: " + getDate(this.deadline) + ")";
+    }
+}
```
Perhaps, instead of leaving the isDone logic in this toString() function, you could leave it in the Task class instead so that you dont have to retype so much code if you need to create a new Task type in the future.

antonlee59 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,150 @@
+        //                 }
+        //                 String[] tempWord = word[1].split("/");
+        //                 String[] from = tempWord[1].split(" ",2);
+        //                 String[] to = tempWord[2].split(" ",2);
+        //                 if (from.length == 1 || to.length == 1) {
+        //                     throw new DukeException("/from and /to needs a date/time.");
+        //                 }
+        //                 try {
+        //                     Task t = new Event(tempWord[0].strip(), from[1].strip(), to[1].strip());
+        //                     lst.add(t);
+        //                     count++;
+        //                     System.out.println("Added new event:\n  " + t + "\nNumber of tasks: " + count);
+        //                     word = br.readLine().strip().split(" ",2);
+        //                 } catch (DateTimeParseException e) {
+        //                     throw new DukeException("Date after /from and /to needs to be in format yyyy-mm-dd");
+        //                 }
+        //             } else if (word[0].equals("delete")) {
+        //                 if (word.length == 1) {
+        //                     throw new DukeException("Delete needs a number.");
+        //                 }
+        //                 if (Integer.parseInt(word[1]) > count) {
+        //                     throw new DukeException("Invalid task.");
+        //                 }
+        //                 Task t = lst.remove(Integer.parseInt(word[1]) - 1);
+        //                 count--;
+        //                 System.out.println("Deleted task:\n  " + t + "\nNumber of tasks: " + count);
+        //                 word = br.readLine().strip().split(" ",2);
+        //             } else {
+        //                 throw new DukeException("Sorry I do not understand the command");
+        //             }
+        //         } catch (DukeException e) {
+        //             System.out.println(e.getMessage());
+        //             word = br.readLine().strip().split(" ",2);
+        //         }
+        //     }
+        //     storeData(lst);
+        //     System.out.println("Duke: Goodbye");
+        // } catch (IOException e) {
+        //     System.out.println(e.getMessage());
+        // }
```
I suggest you to just delete all these unnecessary commented code. That is mainly the point of a version control system like git which is to undo any changes that you might not like in the future. Commenting them out and leaving them in code makes your code unnecssarily long and confusing.

antonlee59 reviewed (commented):
Overall, LGTM. Very clean code and very nice use of ternary operators. Nice abstraction and use of Inheritance to enforce the OOP principles. Good job!

randallnhr committed:
A-Packages

randallnhr committed:
Merge remote-tracking branch 'origin/add-gradle-support'

randallnhr committed:
A-Gradle

randallnhr committed:
A-JUnit

Zhongli5712 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+        taskList.unmarkTask(Integer.parseInt(input[1]) - 1);
+    }
+
+    public void todo(String[] input, TaskList taskList) throws DukeException {
+        if (input.length == 1) {
+            throw new DukeException("todo needs a description");
+        }
+        Task t = new Todo(input[1].strip());
+        taskList.addTask(t);
+        System.out.println("Added new todo:\n  " + t + "\nNumber of tasks: " + taskList.size());
+    }
```
should the method name be "addTodo"?

Zhongli5712 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,56 @@
+import duke.task.Task;
+
+public class TaskList {
+    private ArrayList<Task> taskList;
```
should the name be "tasks"?

Zhongli5712 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    public Parser() {};
+
+    public void readInput(String[] input, TaskList taskList) throws DukeException {
+        switch (input[0]) {
+            case "list":   
+                taskList.list(); 
+                break;        
+            case "mark":
+                mark(input, taskList);
+                break;
+            case "unmark":
+                unmark(input, taskList);
+                break;
+            case "todo":
+                todo(input, taskList);
+                break;
+            case "deadline":
+                deadline(input, taskList);
+                break;
+            case "event":
+                event(input, taskList);
+                break;
+            case "delete":
+                delete(input, taskList);
+                break;
+            default:
+                throw new DukeException("Sorry I do not understand the command");
```
should the switch and case and default be on the same indentation?

Zhongli5712 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    public void event(String[] input, TaskList taskList) throws DukeException {
+        if (input.length == 1 || !input[1].contains("/from") || !input[1].contains("/to") ) {
```
should it be broken down to be three boolean variables?

Zhongli5712 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public String getFileDesc() {
+        return this.isDone
+        ? "D|1|" + this.name + "|" + convertFileDate(this.deadline)
+        : "D|0|" + this.name + "|" + convertFileDate(this.deadline);
+    }
```
I like your break down line for easier reading

Zhongli5712 reviewed (commented):
Overall, I found your code easy to follow, however there are still some minors details that you should notice. Well done!

randallnhr committed:
Add increment A-JavaDoc

randallnhr committed:
Remove redundant codes and correct layout of code

randallnhr committed:
Add increment Level-9

randallnhr committed:
Merge branch 'branch-A-CodingStandard'

randallnhr committed:
Merge branch 'branch-Level-9'

randallnhr committed:
Add GUI

Set up JavaFX and link Duke class to GUI.

Add MainWindow.java as controller for as main GUI.

Add DialogBox.java as controller for DialogBox in GUI.

Add MainWindow.fxml and DialogBox.fxml for GUI layout and interaction

randallnhr committed:
Add GUI

Set up JavaFX and link Duke class to GUI.

Add MainWindow.java as controller for as main GUI.

Add DialogBox.java as controller for DialogBox in GUI.

Add MainWindow.fxml and DialogBox.fxml for GUI layout and interaction

randallnhr committed:
Merge branch 'branch-Level-10'

randallnhr committed:
Add assertions

Assertions make it easier to ensure that certain variables/ methods are
behaving as expected.

By using assertions, we can make debugging easier as assumptions are
ensured.

randallnhr committed:
Merge pull request #2 from randallnhr/branch-A-Assertions

Add assertions

randallnhr committed:
Improve CodeQuality

There are several methods that use complicated expressions.

`mark`, `unmark`, `deadline` and `delete` methods in Parser class has 2
checks in an if condition.
Abstracted the checks to 2 variables to enhance readability.

`event` method in the Parser class has 3 checks in the if condition.
Abstracted the checks to 3 variables to enhance readability.

As a step forward, let's abstract complicated checks in the conditions.

randallnhr committed:
Fix error in Parser

randallnhr committed:
Merge pull request #3 from randallnhr/branch-A-CodeQuality

Branch a code quality

randallnhr committed:
Add reschedule functionality

Allow users to change the /by date of deadline tasks and
/from, /to of event tasks.

randallnhr committed:
Merge branch 'branch-BCD-Extension'

randallnhr committed:
Improve GUI

randallnhr committed:
Add personality

randallnhr committed:
Add personality

randallnhr committed:
Edit delete and error message

randallnhr committed:
Add Ui.png

randallnhr committed:
Edit README.md

randallnhr committed:
Edit README.md
